### PR TITLE
Support for waiting on tasks

### PR DESCRIPTION
**Pull request by richfitz**

### CODE REVIEW

File: R/task.R [hunk 1]
```diff
@@ -384,6 +384,81 @@ task_log_fetch <- function(id, root) {
...
+status_to_logical <- function(status) {
+  switch(status,
+         submitted = NA,
+         running = NA,
+         success = TRUE,
+         failure = FALSE,
+         cancelled = FALSE,
+         cli::cli_abort("Unhandled status '{status}'"))
+}
+
```
Comment (weshinsley) [inline R/task.R:396]: Maybe better naming of this function, to include that success -> TRUE  ?
Is `success_status_as_logical` too wordy?
`is_status_success` maybe?
`final_status_sucess` perhaps hints that if the thing isn't final, it returns NA...

(Only used internally and very locally, so no big deal)

File: R/task.R [hunk 2]
```diff
@@ -384,6 +384,81 @@ task_log_fetch <- function(id, root) {
...
+  path <- file.path(root$path$tasks, id)
+  status <- task_status(id, root = root)
+
+  if (status == "created") {
+    cli::cli_abort(
+      c("Cannot wait on '{id}', which has status '{status}'",
+        i = "You need to submit this task to wait on it"))
+  }
+
+  value <- status_to_logical(status)
```
Comment (weshinsley) [inline R/task.R:434]: Note earlier on possible rename?

File: tests/testthat/test-interface.R [hunk 3]
```diff
@@ -300,8 +299,8 @@ test_that("can read logs", {
   init_quietly(path_there)
   suppressMessages(
     hipercow_configure("elsewhere", path = path_there, root = path_here))
-  suppressMessages(
-    id <- withr::with_dir(path_here, task_create_explicit(quote(sqrt(2)))))
+  suppressMessages(
```
Comment (weshinsley) [inline tests/testthat/test-interface.R:302]:
```suggestion
  suppressMessages(
```

File: tests/testthat/test-interface.R [hunk 4]
```diff
@@ -300,8 +299,8 @@ test_that("can read logs", {
   init_quietly(path_there)
   suppressMessages(
     hipercow_configure("elsewhere", path = path_there, root = path_here))
-  suppressMessages(
-    id <- withr::with_dir(path_here, task_create_explicit(quote(sqrt(2)))))
+  suppressMessages(                                                             
+    id <- withr::with_dir(path_here, task_create_explicit(quote(sqrt(2)))))
```
Comment (weshinsley) [inline tests/testthat/test-interface.R:303]:
```suggestion
    id <- withr::with_dir(path_here, task_create_explicit(quote(sqrt(2)))))
```

**Review by weshinsley — APPROVED** [review]

File: R/task.R (hunk 1 above)
Comment (richfitz) [reply R/task.R:396]: sure, went with `final_status_to_logical` here
